packages/run-protocol/src/psm/psm.js

```diff
@@ -0,0 +1,170 @@
+  from.decrementBy(txFrom);
+  to.incrementBy(txTo);
+}
+
+/**
+ * @param {ZCF<{
+ *    anchorBrand: Brand,
+ *    main: {
+ *      WantStableFeeBP: bigint,
+ *      GiveStableFeeBP: bigint,
+ *      MintLimit: Amount } }>
+ * } zcf
+ * @param {{feeMintAccess: FeeMintAccess}} privateArgs
+ */
+export const start = async (zcf, privateArgs) => {
+  const { anchorBrand } = zcf.getTerms();
+
+  const { feeMintAccess } = privateArgs;
+  // TODO should this know that the name is 'Stable'
+  // TODO get the RUN magic out of here so the contract is more reusable
+  const stableMint = await zcf.registerFeeMint('Stable', feeMintAccess);
+  const { brand: stableBrand } = stableMint.getIssuerRecord();
+  zcf.setTestJig(() => ({
+    stableIssuerRecord: stableMint.getIssuerRecord(),
+  }));
+  const emptyStable = AmountMath.makeEmpty(stableBrand);
+  const emptyAnchor = AmountMath.makeEmpty(anchorBrand);
+
+  // HACK for simple governance API
+  const getGovernance = _ => {
+    const {
+      main: { WantStableFeeBP, GiveStableFeeBP, MintLimit },
+    } = zcf.getTerms();
+    assert(
+      AmountMath.isGTE(MintLimit, emptyAnchor),
+      X`MintLimit ${MintLimit} must specificy a limit in ${anchorBrand}`,
+    );
+    return {
+      getWantStableRate: () =>
+        makeRatio(WantStableFeeBP, stableBrand, BASIS_POINTS),
+      getGiveStableRate: () =>
+        makeRatio(GiveStableFeeBP, stableBrand, BASIS_POINTS),
+      getMintLimit: () => MintLimit,
+    };
+  };
+  const gov = getGovernance(zcf);
+
+  const { zcfSeat: anchorPool } = zcf.makeEmptySeatKit();
+  const { zcfSeat: feePool } = zcf.makeEmptySeatKit();
+  const { zcfSeat: stage } = zcf.makeEmptySeatKit();
+
+  const assertUnderLimit = anchorAmount => {
+    assert(
+      AmountMath.isGTE(gov.getMintLimit(), anchorAmount),
+      X`Request would exceed mint limit`,
+    );
+  };
+
+  const giveStable = (seat, given, wanted = emptyAnchor) => {
+    const fee = ceilMultiplyBy(given, gov.getGiveStableRate());
+    const afterFee = AmountMath.subtract(given, fee);
+    const maxAnchor = AmountMath.make(anchorBrand, afterFee.value);
+    // TODO this prevents the reallocate from failing. Can this be tested otherwise?
+    assert(
+      AmountMath.isGTE(maxAnchor, wanted),
+      X`wanted ${wanted} is more then ${given} minus fees ${fee}`,
+    );
+    stageTransfer(seat, stage, { In: afterFee }, { Stable: afterFee });
+    stageTransfer(seat, feePool, { In: fee }, { Stable: fee });
+    stageTransfer(anchorPool, seat, { Anchor: maxAnchor }, { Out: maxAnchor });
+    zcf.reallocate(seat, anchorPool, stage, feePool);
+    stableMint.burnLosses({ Stable: afterFee }, stage);
+  };
+
+  const wantStable = (seat, given, wanted = emptyStable) => {
+    const anchorAfterTrade = AmountMath.add(
+      anchorPool.getAmountAllocated('Anchor', anchorBrand),
+      given,
+    );
+    assertUnderLimit(anchorAfterTrade);
+    const asStable = AmountMath.make(stableBrand, given.value);
```

Comment:
```suggestion
    const asStable = AmountMath.coerce(stableBrand, given);
```

If `given` isn't trustworthy (it is, Zoe has checked it) then `anchorAfterTrade`  should use asStable rather than given.  If `given` has the wrong brand, the arithmetic and reallocation will fail.

Reply:
This is doing a 1:1 conversion from USDC to RUN. Your point earlier that this needs to use a price even if it is 1:1 because they might have different decimals is spot on though.  So this needs to change to a multiply.

Reply:
It's not a ratio multiply. Much cleaner!